### ISSUE TITLE
Negative values in write pool

### ISSUE THREAD

**Issue by Hitenjain14** [0chain/0chain#2439]
```
allocation_fetch_error: Error fetching the allocation.internal_error: can't get allocation: error retrieving allocation: 212ac818f47f8a095c637c10c9068729f785f384ed4b97da9ee2fa46bc6b0ddf, error: sql: Scan error on column index 32, name "write_pool": converting driver.Value type int64 ("-333644590") to a uint64: invalid syntax
```

**Comment by Sriep** [comment]
This error comes from [GetAllocation](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/dbs/event/allocation.go#L54).

Hence 
```
error: sql: Scan error on column index 32, name "write_pool": converting driver.Value type int64 ("-333644590") to a uint64: invalid syntax
```
is an SQL error trying to a `select *` on the allocations table.

**Comment by dabasov** [comment]
should be something related to uint64 -> int64 conversion

[Transaction value int64 overflow error (postgres bigint doesn't allow unsigned int)](https://github.com/0chain/0chain/issues/2431)

**Comment by peterlimg** [comment]
closed via #2484
